src/app/Plans/components/PlansTable.tsx

```diff
@@ -1,19 +1,119 @@
 import * as React from 'react';
-import { IPlan } from '@app/queries/types';
+import { Button, Level, LevelItem, Pagination, Progress } from '@patternfly/react-core';
+import { StatusIcon, StatusType } from '@konveyor/lib-ui';
+import {
+  Table,
+  TableHeader,
+  TableBody,
+  ICell,
+  IRow,
+  sortable,
+  wrappable,
+} from '@patternfly/react-table';
+
+import PlanStatus from './PlanStatus';
+import PlanActionsDropdown from './PlanActionsDropdown';
+import { useSortState, usePaginationState } from '@app/common/hooks';
+import { IPlan, IMigration } from '@app/queries/types';
 
 interface IPlansTableProps {
-  planList: IPlan[];
-  toggleAddWizardOpen: () => void;
+  plans: IPlan[];
+  migrations: IMigration[];
 }
 
 const PlansTable: React.FunctionComponent<IPlansTableProps> = ({
-  planList,
-  toggleAddWizardOpen,
+  plans,
+  migrations,
 }: IPlansTableProps) => {
+  const getSortValues = (plan: IPlan) => {
+    return [
+      plan.metadata.name,
+      plan.spec.provider.sourceProvider.name,
+      plan.spec.provider.destinationProvider.name,
+      plan.spec.vmList.length,
+      '', // Plan status
+      '', // Action column
+    ];
+  };
+
+  const { sortBy, onSort, sortedItems } = useSortState(plans, getSortValues);
+  const { currentPageItems, setPageNumber, paginationProps } = usePaginationState(sortedItems, 10);
+  React.useEffect(() => setPageNumber(1), [sortBy, setPageNumber]);
+
+  const getMigration = (plan) => migrations.filter((migration) => migration.plan === plan);
+
+  const columns: ICell[] = [
+    { title: 'Name', transforms: [sortable, wrappable] },
+    { title: 'Source provider', transforms: [sortable, wrappable] },
+    { title: 'Target provider', transforms: [sortable] },
+    { title: 'VMs', transforms: [sortable] },
+    { title: 'Plan status', transforms: [sortable] },
+    { title: '' },
+  ];
+
+  const rows: IRow[] = [];
+
+  currentPageItems.forEach((plan: IPlan) => {
+    let buttonText = '';
+    if (plan.status.conditions.every((condition) => condition.type === 'Ready')) {
+      buttonText = 'Start migration';
+    }
+
+    rows.push({
+      meta: { plan },
+      cells: [
+        {
+          title: (
+            <>
+              <a href="#">{plan.metadata.name}</a>
+              <div>{plan.spec.description}</div>
```

Comment:
Also, the plan description text should be a little smaller than the plan name. We can use `<Text component="small">` for that, it has styles which will also make it a shade of gray.